include/ginkgo/core/base/collection.hpp

```diff
@@ -0,0 +1,200 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+#include <numeric>
+
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/base/array.hpp>
+
+
+namespace gko {
+namespace collection {
+
+/**
+ * \brief A minimal std::vector-like interface for a collection of arrays.
+ *
+ * The arrays are stored as sub-views of a single contiguous array.
+ *
+ * \tparam T value type stored in the arrays
+ */
+template <typename T>
+struct array {
```

Comment:
I think I would call this a _segmented array_, since the corresponding operations are referred to a segmented sort/reduction/... in papers and the cub library. That would also avoid the need for a separate namespace